deploy/kubernetes/wait-for-driver.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# This script waits for the deployments from ./deploy-driver.sh to be ready.
+
+print_usage()
+{
+    printf "wait-for-driver.sh [--windows]\n"
+    printf "\t--windows: wait on the windows deployment rather than linux"
+    echo
+}
+
+node_daemonset=csi-gce-pd-node
+while [[ -n "${1-}" ]]; do
+  case $1 in
+    --windows )    shift
+                   node_daemonset=csi-gce-pd-node-win
+                   ;;
+    -h | --help )  print_usage
+                   exit 1
+                   ;;
+    * )            print_usage
+                   exit 1
+                   ;;
+  esac
+done
+
+kubectl wait -n gce-pd-csi-driver deployment csi-gce-pd-controller --for condition=available
```

Comment:
any reason to use kubectl rather than the go client like we do in k/k e2es?

Reply:
The driver is deployed here by shelling out to kustomize and using the deploy scripts. I wanted to be consistent with that, although since the deployment names etc have to be hard-coded into wait-for-driver.sh there's not much benefit except maybe because it lives in the same directory as the deployment scripts, if someone starts changing names around or whatever there's more chance of noticing that file.

It's also turned out to be nice to have a command-line utility that can be used manually with deploy-driver.sh.

I'm not super-attached to this method, though, so if you want to go over to client-go I could do that. It would be slightly annoying to have to set up the client since it's not used elsewhere in the driver but it's not a big deal.

Reply:
Ah I forgot about that. That makes sense then. I wonder if it would be more convenient to put the wait in the deploy-driver.sh directly?

Reply:
IIUC deploy-driver.sh, whether or not we have a windows deployment depends only on if we have windows nodes in the cluster.

In driver.go we have a nice "windows" bool that's easy to check :-)

I think one could have a fancy jsonpath to get that from the node labels on the command line, but I couldn't figure it out.